danger_code_ws/2.cpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <cstdlib>
+
+// Function to dynamically allocate a matrix of size rows x cols
+double** allocateMatrix(int rows, int cols) {
```

Comment:
Consider using `std::vector<std::vector<double>>` instead of manual memory allocation.  This eliminates the risk of memory leaks and simplifies the code significantly.  `std::vector` handles memory management automatically, reducing the chance of errors like forgetting to `deallocateMatrix`.